src/diffusers/training_utils.py

```diff
@@ -251,6 +251,30 @@ def state_dict(self) -> dict:
             "collected_params": self.collected_params,
         }
 
+    def store(self, parameters: Iterable[torch.nn.Parameter]) -> None:
+        r"""
+        Args:
+        Save the current parameters for restoring later.
+            parameters: Iterable of `torch.nn.Parameter`; the parameters to be
+                temporarily stored.
+        """
+        self.collected_params = [param.detach().cpu().clone() for param in parameters]
```

Comment:
Not really sure why we need to put this on CPU and clone. `store`, `restore` shouldn't really take extra memory as we already have both params on GPU. Keeping this on the CPU and then loading it back on GPU will introduce some slowdown.

Reply:
okay play around with it a bit, and think this is okay, sorry about the false alarm.  eval is only once a epoch/few epochs so this shouldn't really be a big issue. Although we should leave a note in the training script that for large models and max perf the user should feel free to modify this.

cc @sayakpaul @patrickvonplaten